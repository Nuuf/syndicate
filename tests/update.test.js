/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('swaps two entities', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({ someData: 0 }));
  syndicate.add(root, syndicate.create({ someData: 0 }));
  syndicate.add(root, syndicate.create({ someData: 0 }));
  const _1 = syndicate.find(root, '1');
  const _2 = syndicate.find(root, '2');
  const _3 = syndicate.find(root, '3');
  _1.data.someData = 1;
  _2.data.someData = 2;
  _3.data.someData = 3;
  syndicate.update(root, _1);
  syndicate.update(root, _2);
  syndicate.update(root, _3);
  expect(JSON.stringify(root, null, 2)).toMatchSnapshot();
});
