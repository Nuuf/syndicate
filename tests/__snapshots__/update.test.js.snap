// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`swaps two entities 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":-1,\\\\\\"parentId\\\\\\":null,\\\\\\"index\\\\\\":0,\\\\\\"id\\\\\\":\\\\\\"root\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"root\\\\\\"},\\\\\\"childrenIndices\\\\\\":[1,2,3],\\\\\\"childrenIds\\\\\\":[\\\\\\"1\\\\\\",\\\\\\"2\\\\\\",\\\\\\"3\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":1,\\\\\\"id\\\\\\":\\\\\\"1\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":1},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"2\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":2},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"3\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":3},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
